### PR TITLE
[stas-molchanov] js practice 2nd layout

### PR DESCRIPTION
**Pull request by moskdev**
https://moskdev.github.io/ui-internship-lwo2/css-practice/task2/index.html

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,4 +10,5 @@ node_modules
 *.log.*
 .DS_STORE
 .DS_Store
-._*
\ No newline at end of file
+._*
```
Comment (AlexLarusso) [inline .gitignore]: Remove. We need sass files with underscore.

File: css-practice/task2/css/style.css [hunk 2]
```diff
@@ -1,7 +1,29 @@
...
+
+.enabled {
+  background-color: #ff6d6d !important;
+  cursor: pointer !important; }
+
+.success {
+  outline: 3px solid green !important; }
+
+.error {
+  outline: 3px solid #ff6d6d !important; }
```
Comment (AlexLarusso) [inline css-practice/task2/css/style.css]: Could you use css-priority instead of !important

File: css-practice/task2/css/style.css [hunk 3]
```diff
@@ -1,7 +1,29 @@
+video {
+  position: absolute;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  z-index: -1;
+  background-color: grey; }
```
Comment (AlexLarusso) [inline css-practice/task2/css/style.css]: Move `}` to next line.

File: css-practice/task2/css/style.css [hunk 4]
```diff
@@ -257,9 +251,6 @@ header img {
 .browser-features {
   background-color: #f6f5f3;
   padding: 0 21.5%;
```
Comment (AlexLarusso) [inline css-practice/task2/css/style.css]: Please set 21.63% should be better. (joke)
Try to don't use decimal values.

File: css-practice/task2/css/style.scss [hunk 5]
```diff
@@ -3,6 +3,34 @@ $color-button-dots: #ff6d6d;
 $color-input-border-and-social: #c8c8c8;
 $color-input: #f6f5f3;
 
+video {
+  position: absolute; 
+  top: 0; left:0;
```
Comment (AlexLarusso) [inline css-practice/task2/css/style.scss]: !

File: css-practice/task2/css/style.scss [hunk 6]
```diff
@@ -3,6 +3,34 @@ $color-button-dots: #ff6d6d;
...
+  width: 100%; height: 100%; 
+  z-index: -1;
+  background-color: grey;
+}
+
+
+
```
Comment (AlexLarusso) [inline css-practice/task2/css/style.scss]: Remove useless empty lines.

File: css-practice/task2/index.html [hunk 7]
```diff
@@ -14,9 +14,10 @@
 
 <body>
   <div class="header-container">
+    <video autoplay src="./img/The-Slow-Dock.mp4"></video>
```
Comment (AlexLarusso) [inline css-practice/task2/index.html]: Use multiple formats (<source>)

File: css-practice/task2/index.html [hunk 8]
```diff
@@ -125,11 +126,11 @@ <h2>Some awesone features</h2>
...
-    <form action="#" class="flexible">
-      <input type="text" placeholder="FULL NAME">
-      <input type="text" placeholder="YOUR EMAIL">
-      <input type="text" placeholder="PASSWORD">
-      <button><img src="./img/30-days.png" alt="30 free days of trial"> TRY NOW</button>
+    <form class="flexible" action="test.php" id="free-trial-form">
+      <input type="text" data-form="name" placeholder="FULL NAME">
+      <input type="email" data-form="email" placeholder="YOUR EMAIL">
+      <input type="password" data-form="password" placeholder="PASSWORD">
+      <button type="submit" class="free-trial-btn"><img src="./img/30-days.png" alt="30 free days of trial"> TRY NOW</button>
```
Comment (AlexLarusso) [inline css-practice/task2/index.html]: use input type submit instead

File: css-practice/task2/script/script.js [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+
+playButton.addEventListener('click', videoToggle);
+trialSubmitForm.addEventListener('keyup', validate, true);
+trialSubmitForm.addEventListener('focus', validate, true);
+trialSubmitForm.addEventListener('keyup', validateForm.bind(null, 'free-trial-form'), true);
+trialSubmitButton.classList.add('disabled');
+
+function validateForm(id, e) {
+  const inputArr = Array.from(document.getElementById(id).children);
+  console.log(inputArr);
```
Comment (AlexLarusso) [inline css-practice/task2/script/script.js]: ?

File: css-practice/task2/script/script.js [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+  for (let i = 0; i < inputArr.length - 1; i++) {
```
Comment (AlexLarusso) [inline css-practice/task2/script/script.js]: Use forEach instead

File: css-practice/task2/script/script.js [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+    console.log(inputArr[i].classList.contains('success'));
```
Comment (AlexLarusso) [inline css-practice/task2/script/script.js]: ?

File: css-practice/task2/script/script.js [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+    e.target.classList.add('error');
+    return false;
+  }
+  if (pattern.test(e.target.value)) {
+    e.target.classList.remove('error');
+    e.target.classList.add('success');
+  } else {
+    e.target.classList.remove('success');
+    e.target.classList.add('error');
+  }
```
Comment (AlexLarusso) [inline css-practice/task2/script/script.js]: DRY.
Simplify please. Think hard.

File: css-practice/task2/script/script.js (hunk 10 above)
Comment (moskdev) [reply css-practice/task2/script/script.js]: my logic won't work the same way , i used it cause i needed to break out of loop if condition is met. Should i rewrite the logic?

File: css-practice/task2/script/script.js (hunk 11 above)
Comment (moskdev) [reply css-practice/task2/script/script.js]: sorry

File: css-practice/task2/script/script.js (hunk 12 above)
Comment (moskdev) [reply css-practice/task2/script/script.js]: fixed:D

File: css-practice/task2/script/script.js (hunk 9 above)
Comment (moskdev) [reply css-practice/task2/script/script.js]: sorry

File: css-practice/task2/index.html (hunk 8 above)
Comment (moskdev) [reply css-practice/task2/index.html]: done

File: css-practice/task2/css/style.scss (hunk 6 above)
Comment (moskdev) [reply css-practice/task2/css/style.scss]: done

File: css-practice/task2/css/style.scss (hunk 5 above)
Comment (moskdev) [reply css-practice/task2/css/style.scss]: done

File: .gitignore (hunk 1 above)
Comment (moskdev) [reply .gitignore]: done

File: css-practice/task2/css/style.css (hunk 2 above)
Comment (moskdev) [reply css-practice/task2/css/style.css]: done

**Pull request merged**